### PR TITLE
Refactor api

### PR DESCRIPTION
**Pull request by jeremylfchen**
Hi @cmonez @Nick-Zabalza ,

Below is my PR for the refactored CRUD routes and testing. I renamed my old `index.js` file to be `datagen.js` so no need to look at that file since it was already reviewed in my previous PR.

- `index.js` now houses all my CRUD operations
- `server.test.js` has been refactored to use my couchDB function calls instead of mongodb

### CODE REVIEW

File: database-couchdb/index.js [hunk 1]
```diff
@@ -1,103 +1,100 @@
...
-const startingItemId = 100;
-const batchSize = 100;
+const { COUCH_URL, COUCH_PWD, COUCH_USER, COUCH_DB } = process.env;
+const url = `http://${COUCH_USER}:${COUCH_PWD}@${COUCH_URL}/${COUCH_DB}`;
+const startId = 100;
 
-const removeOldData = (url, db) => {
+//revision ID helper function needed for couchDB update and delete operations
+//Aka a very specific get request only used to extract the revision id
+const getRevId = (itemId) => {
```
Comment (cmonez) [inline database-couchdb/index.js:13]: I like this helper function! Something that turned me off from couchdb was having to need a revision id to delete?? Seems like an unnecessary extra step, like this function that takes care of that

File: database-couchdb/index.js [hunk 2]
```diff
@@ -1,103 +1,100 @@
...
-      storesStream.destroy();
-      CB(stores);
+    .catch((err) => {
+      console.error(err);
     });
+};
+
+exports.deleteData = async (itemId) => {
 
+  let rev = await getRevId(itemId);
```
Comment (cmonez) [inline database-couchdb/index.js:70]: This looks very easy to read, and I know exactly what's going on, good job writing such readable easy to follow code!

File: database-couchdb/index.js [hunk 3]
```diff
@@ -1,103 +1,100 @@
...
+  //manually assign the new index
+  let id = await axios.get(url)
+    .then((data) => {
+      return data.data.doc_count;
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+
+  return axios.put(url + `/${id + startId}`, data)
```
Comment (cmonez) [inline database-couchdb/index.js:93]: So your waiting for the function above to run before you use this function?

File: server/index.js [hunk 4]
```diff
@@ -31,72 +32,62 @@ app.get('/availableAt/:itemId/', (req, res) => {
...
-//     .then((data) => {
-//       res.status(200).send(data);
-//     })
-//     .catch((err) => {
-//       res.status(500).send(err)
-//     });
-// });
+  updateData(itemId, data)
+    .then((data) => {
+      if (data.ok) {
```
Comment (cmonez) [inline server/index.js:62]: data.ok? What does this mean?

**Review by cmonez — APPROVED** [review]

File: database-couchdb/index.js (hunk 3 above)
Comment (jeremylfchen) [reply database-couchdb/index.js:93]: Yea, not having auto increment really sucks haha

File: server/index.js (hunk 4 above)
Comment (jeremylfchen) [reply server/index.js:62]: The data shape for the response for updates in couch looks like this:
`{
    "id": "blah",
    "ok": true,
    "rev": "2-056f5f44046ecafc08a2bc2b9c229e20"
}`

**Pull request merged**